src/YawPitchControl/YawPitchControl.js

```diff
@@ -276,46 +276,35 @@ const YawPitchControl = class YawPitchControl extends Component {
 		if (keys.some(key => key === "useZoom")) {
 			const useZoom = this.options.useZoom;
 
-			// Init WheelInput
 			// Disconnect first
 			this.axes.disconnect(this.axesWheelInput);
+			this.axesPinchInput && this.axes.disconnect(this.axesPinchInput);
 
 			if (useZoom) {
 				this.axes.connect(["fov"], this.axesWheelInput);
+				this.axesPinchInput && this.axes.connect(["fov"], this.axesPinchInput);
 			} else {
 				this.axes.disconnect(this.axesWheelInput);
+				this.axesPinchInput && this.axes.disconnect(this.axesPinchInput);
 			}
 
-			// Init PinchInput
-			if (this.axesPinchInput) {
-				// Disconnect first
-				this.axes.disconnect(this.axesPinchInput);
-
-				if (useZoom) {
-					this.axes.connect(["fov"], this.axesPinchInput);
-				} else {
-					this.axes.disconnect(this.axesPinchInput);
-				}
+			this._shouldTogglePinchInput = true;
+		}
 
-				keys.push("touchDirection");
-			}
+		if (keys.some(key => key === "touchDirection") || this._shouldTogglePinchInput) {
+			this._togglePinchInputByOption(this.options.touchDirection, this.options.useZoom);
+			this._shouldTogglePinchInput = false;
 		}
 
 		if (keys.some(key => key === "touchDirection")) {
 			this._enabled && this._enableTouch(this.options.touchDirection);
 		}
 	}
 
-	_enableTouch(direction) {
-		// Disconnect first
-		this.axesPanInput && this.axes.disconnect(this.axesPanInput);
-
-		const yawEnabled = direction & TOUCH_DIRECTION_YAW ? "yaw" : null;
-		const pitchEnabled = direction & TOUCH_DIRECTION_PITCH ? "pitch" : null;
-
+	_togglePinchInputByOption(touchDirection, useZoom) {
 		// If the touchDirection option is not ALL, pinchInput should be disconnected to make use of a native scroll.
-		if (this.axesPinchInput && this.options.useZoom) {
-			if (direction === TOUCH_DIRECTION_ALL) {
+		if (this.axesPinchInput && useZoom) {
+			if (touchDirection === TOUCH_DIRECTION_ALL) {
```

Comment:
@happyhj How about following?
It can reduce the `if` depth and make more readability.

```js
if (this.axesPinchInput && useZoom && touchDirection === TOUCH_DIRECTION_ALL) {
  // TODO: Get rid of using private property of axes instance.
  if (this.axes._inputs.indexOf(this.axesPinchInput) === -1) {
    this.axes.connect(["fov"], this.axesPinchInput);
  }
} else {
  this.axes.disconnect(this.axesPinchInput);
}
```

Reply:
@happyhj and the flag is not needed anymore.

How about calling `togglePinchInputByOption` always..because `togglePinchInputByOption` check condition by  itself.
So. instead of 
```js
if (keys.some(key => key === "touchDirection") || this._shouldTogglePinchInput) {
  this._togglePinchInputByOption(this.options.touchDirection, this.options.useZoom);
  this._shouldTogglePinchInput = false;
}
```
just call

```js
this._togglePinchInputByOption(this.options.touchDirection, this.options.useZoom);
```